.github/workflows/generate-data.yaml

```diff
@@ -88,214 +96,187 @@ jobs:
           repository: ${{ steps.id.outputs.repo }}
           sparse-checkout-cone-mode: false
           sparse-checkout: |
+            predtimechart-config.yml
             predevals-config.yml
             site-config.yml
-      - id: status-check
+          path: 'cfg/'
+      - id: get-hub-name
+        name: "Get information about hub"
+        run: |
+          yq eval '.hub' cfg/site-config.yml \
+          | sed -E -e 's+(https://github.com/|^)+hub=+' \
+          | sed -E -e 's+[/]$++' >> $GITHUB_OUTPUT
+      - id: status
         name: "Check if we need to build data"
         run: |
-          if [[ ! -f predevals-config.yml ]]; then
-            echo "ok=false" >> "$GITHUB_OUTPUT"
-            repo=${{ steps.id.outputs.repo }}
-            echo "::notice title=Missing config ($repo)::No predevals-config.yml file found. Exiting."
-            exit 0
+          repo=${{ steps.id.outputs.repo }}
+          if [[ ! -f cfg/predevals-config.yml ]]; then
+            echo "eval-ok=false" >> "$GITHUB_OUTPUT"
+            echo "::notice title=$repo: No evals to build::No predevals-config.yml file found."
+          else
+            echo "eval-ok=true" >> $GITHUB_OUTPUT
           fi
-          file=predevals-${{ env.KEY }}
-          touch "$file"
-          {
-            echo '${{ toJSON(matrix.site) }}'
-          } > "$file"
-          echo "ok=true" >> $GITHUB_OUTPUT
-      - id: upload-status
-        if: ${{ fromJSON(steps.status-check.outputs.ok) }}
+          if [[ ! -f cfg/predtimechart-config.yml ]]; then
+            echo "forecast-ok=false" >> "$GITHUB_OUTPUT"
+            echo "::notice title=$repo: No forecasts to build::No predtimechart-config.yml file found."
+          else
+            echo "forecast-ok=true" >> $GITHUB_OUTPUT
+          fi
+      - name: Upload artifact
+        if: ${{ fromJSON(steps.status.outputs.forecast-ok) || fromJSON(steps.status.outputs.eval-ok) }}
+        id: upload
         uses: actions/upload-artifact@v4
         with:
-          name: predevals-${{ env.KEY }}
-          path: predevals-${{ env.KEY }}
+          name: ${{ steps.id.outputs.key }}-cfg
+          path: 'cfg/'
+
+  build-evals:
+    needs: [check]
+    if: ${{ fromJSON(needs.check.outputs.eval-ok) }}
```

Comment:
this if statement controls this entire job and prevents it from running if no evals can be built.